server/src/main/java/io/spine/server/projection/ProjectionRepository.java

```diff
@@ -91,6 +92,28 @@ EventStore getEventStore() {
     }
 
     /**
+     * {@inheritDoc}
+     *
+     * <p>Ensures there is at least one event subscriber method (external or domestic) declared
+     * by the class of the projection. Throws an {@code IllegalStateException} otherwise.
+     */
+    @Override
+    public void onRegistered() {
+        super.onRegistered();
+
+        final boolean noEventSubscriptions = getMessageClasses().isEmpty();
+        if (noEventSubscriptions) {
+            final boolean noExternalSubscriptions = getExternalEventDispatcher().getMessageClasses()
+                                                                                .isEmpty();
+            if (noExternalSubscriptions) {
+                throw newIllegalStateException(
+                        "Projections of the repository %s have neither domestic nor external " +
```

Comment:
Let's have `Projections of type ...` instead `of repository ...`. It seems more natural.

Reply:
As discussed, this an emphasis of a **repository**, which validation mechanism failed to complete a process of _Repository-in-Bounded-Context Registration_. 

So to me it makes sense.